app/models/agents/weather_agent.rb

```diff
@@ -111,33 +125,36 @@ def validate_location
         errors.add(
           :base,
           "Location #{location} is malformed. Location for " +
-          'Dark Sky must be in the format "-00.000,-00.00000". The ' +
+          'OpenWeather must be in the format "-00.000,-00.00000". The ' +
           "number of decimal places does not matter.")
       end
     end
 
     def validate_options
-      errors.add(:base, "The Weather Underground API has been disabled since Jan 1st 2018, please switch to DarkSky") if wunderground?
+      errors.add(:base, "The DarkSky API has been disabled since Aug 1st, 2020; please switch to OpenWeather.") if darksky?
+      errors.add(:base, "The Weather Underground API has been disabled since Jan 1st 2018; please switch to OpenWeather.") if wunderground?
       validate_location
       errors.add(:base, "api_key is required") unless interpolated['api_key'].present?
       errors.add(:base, "which_day selection is required") unless which_day.present?
     end
 
-    def dark_sky
+    def open_weather
       if key_setup?
-        ForecastIO.api_key = interpolated['api_key']
+        onecall_endpoint = "http://api.openweathermap.org/data/2.5/onecall"
```

Comment:
It sounds like new OpenWeatherMap account cannot use this API anymore. Do you think we can use https://api.openweathermap.org/data/2.5/weather instead? Doc: https://openweathermap.org/current